### PR TITLE
Add release builds to TravisCI build matrix

### LINKED ISSUE #233: Add release builds to macOS/Linux CI
I closed #141 a little early. We should add release builds to macOS/Linux too.

### PR DESCRIPTION
**Pull request by mborgerson**
Adds a configuration to build release builds for Linux/macOS platforms. My motivation for this is mostly to capture any build errors typically introduced with debug macro guards. We also will likely add a deployment step in the future for people to download the release builds--this gets that started.

Also adds badges to README.md to show the status of these jobs. I faked it for the status of the Appveyor job (it should be a logical AND of both debug and release builds). I'm not sure how to show the badge for the individual job on Appveyor yet.

Closes #233

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,6 +6,7 @@ and Linux. Please visit [xqemu.com](http://xqemu.com) to learn more.
 
 Build Status
 ------------
-| Windows | macOS | Linux |
-| ------- | ----- | ----- |
-| [![Build status](https://ci.appveyor.com/api/projects/status/i1m0keigjabxg170/branch/master?svg=true)](https://ci.appveyor.com/project/xqemu-bot/xqemu) | [![Travis-CI Status](https://travis-matrix-badges.herokuapp.com/repos/xqemu/xqemu/branches/master/2)](https://travis-ci.org/xqemu/xqemu) | [![Travis-CI Status](https://travis-matrix-badges.herokuapp.com/repos/xqemu/xqemu/branches/master/1)](https://travis-ci.org/xqemu/xqemu) |
+| Build | Windows | macOS | Linux |
+| ----- | ------- | ----- | ----- |
+| Debug | [![Build status](https://ci.appveyor.com/api/projects/status/i1m0keigjabxg170/branch/master?svg=true)](https://ci.appveyor.com/project/xqemu-bot/xqemu/branch/master) | [![Travis-CI Status](https://travis-matrix-badges.herokuapp.com/repos/xqemu/xqemu/branches/master/3)](https://travis-ci.org/xqemu/xqemu) | [![Travis-CI Status](https://travis-matrix-badges.herokuapp.com/repos/xqemu/xqemu/branches/master/1)](https://travis-ci.org/xqemu/xqemu) |
```
Comment (JayFoxRox) [inline README.md]: nit: Table padding with spaces would have been nice to make it easier to compare debug and release lines:

```
| Build   | Windows | macOS | Linux |
| ------- | ------- | ----- | ----- |
| Debug   | [![Build status](https://ci.appveyor.com/api/projects/status/i1m0keigjabxg170/branch/master?svg=true)](https://ci.appveyor.com/project/xqemu-bot/xqemu/branch/master) | [![Travis-CI Status](https://travis-matrix-badges.herokuapp.com/repos/xqemu/xqemu/branches/master/3)](https://travis-ci.org/xqemu/xqemu) | [![Travis-CI Status](https://travis-matrix-badges.herokuapp.com/repos/xqemu/xqemu/branches/master/1)](https://travis-ci.org/xqemu/xqemu) |
| Release | [![Build status](https://ci.appveyor.com/api/projects/status/i1m0keigjabxg170/branch/master?svg=true)](https://ci.appveyor.com/project/xqemu-bot/xqemu/branch/master) | [![Travis-CI Status](https://travis-matrix-badges.herokuapp.com/repos/xqemu/xqemu/branches/master/4)](https://travis-ci.org/xqemu/xqemu) | [![Travis-CI Status](https://travis-matrix-badges.herokuapp.com/repos/xqemu/xqemu/branches/master/2)](https://travis-ci.org/xqemu/xqemu) |
```

Comment (JayFoxRox) [inline README.md]: nit: I think release should be on top; release is our primary goal so it should be in top left.

*(However, on xqemu.com / ecosystem, we should still only promote debug builds for now; I consider this release-check only a development aid for now)*

Comment (JayFoxRox) [inline README.md]: nit: Somehow labels for the Travis and AppVeyor images are different:

`![Build status]` vs. `![Travis-CI Status]`

We should make this more consistent while you are at it.

Comment (JayFoxRox) [reply README.md]: I force pushed it

Comment (mborgerson) [reply README.md]: Approved. Go for merge.

**Pull request merged**